lib/siteAdapter.js

```diff
@@ -7,6 +7,36 @@ const localForage = require('localforage')
 
 module.exports = siteAdapter = {}
 
+// IPv6/localhost-safe URL helpers -------------------------------------------
+const isLoopbackHost = h =>
+  h === 'localhost' || h === '127.0.0.1' || h === '::1' || h === '[::1]';
```

Comment:
This needs to cater for localhost subdomains.

Which is what `if (site.split('.').at(-1).split(':')[0] === 'localhost') {` is part of.